qiita_db/support_files/patches/33.sql

```diff
@@ -0,0 +1,435 @@
+
+-- Parent artifact table - keeps track of the procenance of a given artifact.
+-- If an artifact doesn't have a parent it means that it was uploaded by the user.
+CREATE TABLE qiita.parent_artifact (
+    artifact_id          bigint  NOT NULL,
+    parent_id            bigint  NOT NULL,
+    CONSTRAINT idx_parent_artifact PRIMARY KEY ( artifact_id, parent_id )
+ ) ;
+CREATE INDEX idx_parent_artifact_artifact ON qiita.parent_artifact ( artifact_id ) ;
+CREATE INDEX idx_parent_artifact_parent ON qiita.parent_artifact ( parent_id ) ;
+ALTER TABLE qiita.parent_artifact ADD CONSTRAINT fk_parent_artifact_artifact FOREIGN KEY ( artifact_id ) REFERENCES qiita.artifact( artifact_id )    ;
+ALTER TABLE qiita.parent_artifact ADD CONSTRAINT fk_parent_artifact_parent FOREIGN KEY ( parent_id ) REFERENCES qiita.artifact( artifact_id )    ;
+
+-- Study artifact table - relates each artifact with its study
+CREATE TABLE qiita.study_artifact (
+    study_id             bigint  NOT NULL,
+    artifact_id          bigint  NOT NULL,
+    CONSTRAINT idx_study_artifact PRIMARY KEY ( study_id, artifact_id )
+ ) ;
+CREATE INDEX idx_study_artifact_study ON qiita.study_artifact ( study_id ) ;
+CREATE INDEX idx_study_artifact_artifact ON qiita.study_artifact ( artifact_id ) ;
+ALTER TABLE qiita.study_artifact ADD CONSTRAINT fk_study_artifact_study FOREIGN KEY ( study_id ) REFERENCES qiita.study( study_id )    ;
+ALTER TABLE qiita.study_artifact ADD CONSTRAINT fk_study_artifact_artifact FOREIGN KEY ( artifact_id ) REFERENCES qiita.artifact( artifact_id )    ;
+
+-- Software table - holds the information of a given software package present
+-- in the system and can be used to process an artifact
+CREATE TABLE qiita.software (
+    software_id          bigserial  NOT NULL,
+    name                 varchar  NOT NULL,
+    version              varchar  NOT NULL,
+    description          varchar  NOT NULL,
+    CONSTRAINT pk_software PRIMARY KEY ( software_id )
+ ) ;
+
+-- soft_command table - holds the information of a command in a given software
+-- this table should be renamed to command once the command table in the
+-- analysis table is merged with this one
+CREATE TABLE qiita.soft_command (
+    command_id           bigserial  NOT NULL,
+    name                 varchar  NOT NULL,
+    software_id          bigint  NOT NULL,
+    description          varchar  NOT NULL,
+    cli_cmd              varchar  ,
+    parameters_table     varchar  NOT NULL,
+    CONSTRAINT pk_soft_command PRIMARY KEY ( command_id )
+ ) ;
+CREATE INDEX idx_soft_command ON qiita.soft_command ( software_id ) ;
+ALTER TABLE qiita.soft_command ADD CONSTRAINT fk_soft_command_software FOREIGN KEY ( software_id ) REFERENCES qiita.software( software_id );
+
+-- Publication table - holds the minimum information for a given publication
+-- It is useful to keep track of the publication of the studies and the software
+-- used for processing artifacts
+CREATE TABLE qiita.publication (
```

Comment:
What do you gain by having two tables over one though? From that conversation I'm not sure it's necessary. You can PK on all 3 columns instead of just the two, and you can do the data move gracefully as you don't need a DOI to attach a PMID to.